api/fastify-app.js

```diff
@@ -0,0 +1,70 @@
+import Fastify from 'fastify'
+import * as Sentry from '@sentry/node'
+import { createRoutes } from './routes/index.js'
+
+export async function createFastifyApp({
+  pgPool,
+  logger,
+  dealIngestionAccessToken,
+  domain
+}) {
+  const fastify = Fastify({
+    logger: {
+      level: 'info',
+      transport: {
+        target: 'pino-pretty'
+      }
+    }
+  })
+
+  // Add shared context to each request
+  fastify.decorateRequest('pgPool', null)
+  fastify.decorateRequest('dealIngestionAccessToken', null)
+  fastify.decorateRequest('domain', null)
+  
+  // Add hooks to setup request context
+  fastify.addHook('onRequest', async (request, reply) => {
+    request.pgPool = pgPool
+    request.dealIngestionAccessToken = dealIngestionAccessToken
+    request.domain = domain
+
+    // Domain redirect check
+    if (request.headers.host && request.headers.host.split(':')[0] !== domain) {
+      return reply.redirect(301, `https://${domain}${request.url}`)
+    }
+  })
+
+  // Error handler
+  fastify.setErrorHandler((error, request, reply) => {
```

Comment:
Can we use fastify's default error handler instead? No need to keep the current implementation, which we only created, because we didn't have any default error handler